### ISSUE TITLE
(void-variable git-commit-mode-map) in magit-status

### ISSUE THREAD

**Issue by tpapp** [magit/magit#5225]
I think this is caused by an update a couple of weeks ago, but I was not using magit at the time so I did not notice. Now, with `magit-20240912.1140`, Emacs 29.3, `Magit 20240912.1140 [>= 4.1.0], Transient 0.7.5, Git 2.43.0, Emacs 29.3, gnu/linux`, I get the following backtrace after calling `magit-status`:

```
Debugger entered--Lisp error: (void-variable git-commit-mode-map)
  (keymap-set git-commit-mode-map "C-c C-d" 'magit-diff-while-committing)
  require(magit-diff)
  byte-code("\300\301!\210\300\302!\210\300\303!\210\300\304!\210\300\305!\210\300\306!\210\300\307!\210\300\310!\210\300\311\312\313#\210\300\314!\210\315\316\317\320\321DD\322..." [require magit-core magit-diff magit-log magit-wip magit-apply magit-repos git-commit format-spec package nil t with-editor custom-declare-variable magit-openpgp-default-signing-key funcall function #f(compiled-function () #<bytecode 0x19800016fe914>) "Fingerprint of your default Openpgp key used for s..." :package-version (magit . "4.0.0") :group magit-commands :type string] 10)
  require(magit)
  byte-code("\301\302!\210\303\304\305\306\307\310\311\312&\7\210\313\314\315\316\317DD\320\311\304\321\322&\7\210\313\323\315\316\324DD\325\326\327\311\304\321\322\330\331&\13\210\313..." [magit-log-margin--custom-type require magit custom-declare-group magit-status nil "Inspect and manipulate Git repositories." :link (info-link "(magit)Status Buffer") :group magit-modes custom-declare-variable magit-status-mode-hook funcall function #f(compiled-function () #<bytecode 0x19800016fe914>) "Hook run after entering Magit-Status mode." :type hook magit-status-headers-hook #f(compiled-function () #<bytecode 0x1e90d0c51a9a14>) "Hook run to insert headers into the status buffer...." :package-version (magit . "2.1.0") :options (magit-insert-error-header magit-insert-diff-filter-header magit-insert-repo-header magit-insert-remote-header magit-insert-head-branch-header magit-insert-upstream-branch-header magit-insert-push-branch-header magit-insert-tags-header) magit-status-sections-hook #f(compiled-function () #<bytecode 0x1358cf05789f0f14>) "Hook run to insert sections into a status buffer." (magit . "2.12.0") magit-status-initial-section #f(compiled-function () #<bytecode 0x19800016fe854>) "The section point is placed on when a status buffe..." (magit . "2.90.0") (choice (const :tag "as usual" nil) (repeat (choice (number :tag "nth top-level section") (sexp :tag "section identity")))) magit-status-goto-file-position #f(compiled-function () #<bytecode 0x19800016fe914>) "Whether to go to position corresponding to file po..." (magit . "3.0.0") boolean magit-status-show-hashes-in-headers #f(compiled-function () #<bytecode 0x19800016fe914>) "Whether headers in the status buffer show hashes.\n..." (magit . "2.4.0") magit-status-margin #f(compiled-function () #<bytecode 0x1e66b2a3a00d0ab5>) "Format of the margin in `magit-status-mode' buffer..." (magit . "2.9.0") magit-margin :initialize ...] 20)
  command-execute(magit-status)
```

I *did* uninstall magit/reinstall, and **that fixes the issue until I restart Emacs**. 

If I run the command from `magit-emacs-Q-command`, the issue **does not show up**.

I am suspecting this is some kind of byte-compile glitch, but don't know how to debug this further.

**Comment by tpapp** [comment]
Removing `.emacs.d/elpa/**/*.elc` did not fix the issue, but nuking the whole `.emacs.d/elpa/` did. Closing, sorry for the noise.

**Comment by tarsius** [comment]
> Removing `.emacs.d/elpa/**/*.elc` did not fix the issue, but nuking the whole `.emacs.d/elpa/` did. Closing, sorry for the noise.

I am guessing that the latter workd while the former did not, because it ensured that the old `.emacs.d/elpa/git-commit-*/` got removed. `git-commit.el` is no longer distributed as a separate package. Instead it is included in the `magit` package.

(Though I don't really understand why a leftover `git-commit` package would cause this issue.)

So for other people running into this issue: before nuking `~/.emacs.d/elpa/`, try removing just `~/.emacs.d/elpa/git-commit-*/`.

Finally, in 3e755c48f2e85090043586a2e8ddeba964d9a11d I have removed the `keymap-set` instance it triggered this issue. That will be included in the next release, coming in two weeks.
